tests/tests/coins.rs

```diff
@@ -165,7 +165,7 @@ mod coin {
             .await
             .unwrap();
         assert_eq!(coins_per_asset.len(), 2);
-        assert!(coins_per_asset[0].len() >= 1);
+        assert!(coins_per_asset[0].is_empty());
```

Comment:
is this a functional change in your pr?

Reply:
I guess it's the `clippy::len_zero` lint kicking in, probably due to using a newer toolchain.

And the updated assert should say:
```suggestion
        assert!(!coins_per_asset[0].is_empty());
```

Reply:
when I ran `cargo clippy --all-targets`, suggested those changes to pass all checks